optuna_integration/trackio/trackio.py

```diff
@@ -0,0 +1,333 @@
+            visualizations, and sweep-style dashboards.
+
+            If ``False`` (default), all trials are logged into a single run.
+
+        space_id:
+            Optional Hugging Face Space ID (``"username/space-name"``) to
+            which the tracked project will be synchronized. If the Space
+            does not exist, it will be created automatically.
+
+        dataset_id:
+            Optional Hugging Face Dataset ID (``"username/dataset-name"``)
+            used to export trial metrics and parameters as a versioned
+            dataset for offline analysis and reproducibility.
+
+        private:
+            Whether the Hugging Face Space or Dataset should be private.
+            Defaults to the user or organization’s Hub settings.
+
+        trackio_kwargs:
+            Additional keyword arguments passed to :func:`trackio.init`,
+            such as ``resume`` or UI-related configuration options.
+    """
+
+    def __init__(
+        self,
+        project: str,
+        metric_name: str | Sequence[str] = "value",
+        *,
+        as_multirun: bool = False,
+        space_id: str | None = None,
+        dataset_id: str | None = None,
+        private: bool | None = None,
+        trackio_kwargs: dict[str, Any] | None = None,
+    ) -> None:
+        _imports.check()
+
+        if not isinstance(metric_name, (str, Sequence)):
+            raise TypeError(f"metric_name must be str or sequence[str], got {type(metric_name)}")
+
+        self._project = project
+        self._metric_name = metric_name
+        self._as_multirun = as_multirun
+        self._space_id = space_id  # HF inits for trackio
+        self._dataset_id = dataset_id
+        self._private = private
+        self._trackio_kwargs = trackio_kwargs or {}
+
+        # Explicit internal state (DO NOT infer from Trackio, will cause errors)
+        self._objective_wrapped: bool = False
+        self._base_run_name: str | None = None
+        self._active_trial_number: int | None = None
+
+    def __call__(
+        self,
+        _study: optuna.study.Study,
+        trial: optuna.trial.FrozenTrial,
+    ) -> None:
+        if self._base_run_name is None:
+            self._base_run_name = _study.study_name
+        if trial.values is None:
+            return
+
+        # If the objective was not wrapped, we cannot safely log
+        if not self._objective_wrapped:
+            if self._as_multirun:
+                print(
+                    "TrackioCallback(as_multirun=True) requires the objective to be "
+                    "wrapped with @trackioc.track_in_trackio(). "
+                )
+            return
+
+        metrics = self._build_metrics(trial)
+
+        # Safe wrapper guarantees a live Trackio run
+        trackio.log(
+            {
+                **trial.params,
+                **metrics,
+                "trial_number": trial.number,
+            },
+            step=trial.number,
+        )
+
+    @experimental_func("4.7.0")
```

Comment:
Since `TrackioCallback` class is marked with the `experimental_class` decorator, I think we can remove the `experimental_func` decorator to avoid issuing multiple warning messages.

```suggestion
```